app/src/test/java/com/lastaoutdoor/lasta/data/db/ActivitiesDBRepositoryImplTest.kt

```diff
@@ -0,0 +1,296 @@
+package com.lastaoutdoor.lasta.data.db
+
+import android.content.Context
+import com.google.firebase.firestore.CollectionReference
+import com.google.firebase.firestore.DocumentReference
+import com.google.firebase.firestore.DocumentSnapshot
+import com.google.firebase.firestore.FieldPath
+import com.google.firebase.firestore.FirebaseFirestore
+import com.google.firebase.firestore.Query
+import com.google.firebase.firestore.QuerySnapshot
+import com.lastaoutdoor.lasta.R
+import com.lastaoutdoor.lasta.data.MockTask
+import com.lastaoutdoor.lasta.models.activity.Activity
+import com.lastaoutdoor.lasta.models.activity.ActivityType
+import com.lastaoutdoor.lasta.models.activity.ClimbingStyle
+import com.lastaoutdoor.lasta.models.activity.Difficulty
+import com.lastaoutdoor.lasta.models.activity.Rating
+import com.lastaoutdoor.lasta.models.api.Position
+import io.mockk.clearAllMocks
+import io.mockk.coEvery
+import io.mockk.coVerify
+import io.mockk.every
+import io.mockk.mockk
+import kotlinx.coroutines.test.runTest
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+
+class ActivitiesDBRepositoryImplTest {
+
+  private lateinit var activitiesDB: ActivitiesDBRepositoryImpl
+  private val database: FirebaseFirestore = mockk(relaxed = true)
+  private val context: Context = mockk()
+  private val collection: CollectionReference = mockk(relaxed = true)
+  private val documentReference: DocumentReference = mockk()
+  private val documentSnapshot1: DocumentSnapshot = mockk()
+  private val documentSnapshot2: DocumentSnapshot = mockk()
+  private val query: Query = mockk()
+  private val querySnapshot: QuerySnapshot = mockk()
+  private val addTask: MockTask<DocumentReference> = mockk()
+  private val getTask: MockTask<DocumentSnapshot> = mockk()
+  private val queryTask: MockTask<QuerySnapshot> = mockk()
+  private val updateTask: MockTask<Void> = mockk()
+
+  @Before

```

Comment:
Tests Work, but might want to declare more stuff inside tests instead of in setUp